src/lib.rs

```diff
@@ -1,13 +1,31 @@
-//! # num_cpus
-//!
 //! A crate with utilities to determine the number of CPUs available on the
 //! current system.
+//! 
+//! Sometimes the cpu will exaggerate the number of cpus it contains, because it can use
+//! [processor tricks] to deliver increased performance when there are more threads. This 
+//! crate provides methods to get both the logical and physical numbers of cores.
+//! 
+//! This information can be used as a guide to how many tasks can be run in parallel.
+//! There are many properties of the system architecture that will affect parallelism,
+//! for example memory access speeds (for all the caches and RAM) and the physical
+//! architecture of the processor, so the number of CPUs should be used as a rough guide
+//! only.
+//! 
 //!
-//! ## Example
+//! ## Examples
 //!
+//! Fetch the number of logical cpus.
+//! 
 //! ```
 //! let cpus = num_cpus::get();
 //! ```
+//! 
+//! See [`rayon::Threadpool`] for an example of where the number of cpus could be
+//! used when setting up parallel jobs (Where the threadpool example uses a fixed
+//! number 8, it could use the number of cpus).
+//! 
+//! [processor tricks]: https://en.wikipedia.org/wiki/Simultaneous_multithreading
+//! [`rayon::ThreadPool`]: https://doc.rs/rayon/0.8.2/rayon/struct.ThreadPool.html
```

Comment:
Link is broken, should be https://docs.rs/rayon/0.8.2/rayon/struct.ThreadPool.html (instead of doc.rs)